var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -675,7 +713,28 @@ def define_tpl_enable(cmake_var, spec_var=None):
 
         options.append(self.define_from_variant('TPL_ENABLE_ADIOS2', 'adios2'))
 
+        options.append(define(
+            f"Kokkos_ARCH_{Kokkos.spack_micro_arch_map[spec.target.name].upper()}",
```

Comment:
This is the line that's not compatible with python 2 (or < 3.6ish I think). Just concatenate the two strings.

Reply:
fixed